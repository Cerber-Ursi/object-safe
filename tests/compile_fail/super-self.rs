use object_safe::object_safe;

trait Super<T: ?Sized> {
    fn gen(_: T);
}

#[object_safe]
trait Test: Super<Self> {}

#[object_safe]
trait Test2: Debug + Super<Self> + Copy {}

#[object_safe]
trait Test3: Super<dyn Fn(Self)> {}

fn main() {}
