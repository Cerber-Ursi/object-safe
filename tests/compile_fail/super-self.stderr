error: Trait with Self in supertrait generic parameters cannot be made object-safe
 --> $DIR/super-self.rs:8:13
  |
8 | trait Test: Super<Self> {}
  |             ^^^^^^^^^^^

error: Trait with Self in supertrait generic parameters cannot be made object-safe
  --> $DIR/super-self.rs:11:22
   |
11 | trait Test2: Debug + Super<Self> + Copy {}
   |                      ^^^^^^^^^^^

error: Trait with Self in supertrait generic parameters cannot be made object-safe
  --> $DIR/super-self.rs:14:14
   |
14 | trait Test3: Super<dyn Fn(Self)> {}
   |              ^^^^^^^^^^^^^^^^^^^
